addons/sale_timesheet/models/account.py

```diff
@@ -49,9 +49,11 @@ def _compute_commercial_partner(self):
         for timesheet in self:
             timesheet.commercial_partner_id = timesheet.task_id.partner_id.commercial_partner_id or timesheet.project_id.partner_id.commercial_partner_id
 
-    @api.depends('so_line.product_id', 'project_id', 'amount')
+    @api.depends('so_line.product_id', 'project_id.billing_type', 'amount')
     def _compute_timesheet_invoice_type(self):
-        for timesheet in self:
+        billed_manually_timesheets = self.filtered(lambda timesheet: timesheet.project_id.allow_billable and timesheet.project_id.billing_type == 'manually')
+        billed_manually_timesheets.timesheet_invoice_type = 'billable_manual'
+        for timesheet in self - billed_manually_timesheets:
             if timesheet.project_id:  # AAL will be set to False
                 invoice_type = 'non_billable' if not timesheet.so_line else False
```

Comment:
Or we should just change that line to set `billable_manual` if the timesheet has no SOL and the task linked has no SOL and the billing_type of the project linked is set to manually. 🤔